### PR TITLE
Renaming delta-kernel-python to delta-kernel-rust-sharing-wrapper

### PR DESCRIPTION
**Pull request by pranavsuku-db**

### CODE REVIEW

File: python/delta-kernel-rust-sharing-wrapper/Cargo.toml [hunk 1]
```diff
@@ -1,11 +1,11 @@
 [package]
-name = "delta-kernel-python"
+name = "delta-kernel-rust-sharing-wrapper"
 edition = "2021"
 license = "Apache-2.0"
-version = "0.0.0"
+version = "0.0.1"
```
Comment (linzhou-db) [inline python/delta-kernel-rust-sharing-wrapper/Cargo.toml:5]: let's still test with 0.0.1 and in the actual release we'll use 0.1.0

File: dev/python_release.sh [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+#git clean -fdx
+
+cd python
+cd delta-kernel-rust-sharing-wrapper
+rm -Rf .venv
+python3 -m venv .venv
+source .venv/bin/activate
+pip install maturin
+maturin build
+
```
Comment (linzhou-db) [inline dev/python_release.sh:17]: L10 - L17 seems not needed if we are using github actions to release the sharing wrapper.

**Review by linzhou-db — APPROVED** [review]

File: python/delta-kernel-rust-sharing-wrapper/Cargo.toml (hunk 1 above)
Comment (pranavsuku-db) [reply python/delta-kernel-rust-sharing-wrapper/Cargo.toml:5]: makes sense

File: dev/python_release.sh (hunk 2 above)
Comment (pranavsuku-db) [reply dev/python_release.sh:17]: yeah ill delete this

File: dev/python_release.sh [hunk 3]
```diff
@@ -0,0 +1,33 @@
+#!/bin/bash -e -pipe
+
+# Switch to the project root directory
+cd $( dirname $0 )
+cd ..
+
+# Clean up uncommitted files
```
Comment (linzhou-db) [inline dev/python_release.sh:7]: and we can remove these two lines.

**Review by linzhou-db — APPROVED** [review]

File: python/setup.py [hunk 4]
```diff
@@ -78,11 +76,12 @@
...
-        'Programming Language :: Python :: 3.7',
         'Programming Language :: Python :: 3.8',
         'Programming Language :: Python :: 3.9',
+        'Programming Language :: Python :: 3.10',
+        'Programming Language :: Python :: 3.11',
     ],
 )
 
-chdir('delta-kernel-python')
+chdir('delta-kernel-rust-sharing-wrapper')
```
Comment (linzhou-db) [inline python/setup.py:86]: is L86 and L87 still needed for sth?

**Review by linzhou-db — APPROVED** [review]

File: python/setup.py (hunk 4 above)
Comment (pranavsuku-db) [reply python/setup.py:86]: nope let me remove

**Pull request merged**